pkg/deploy/lattice/target_group_synthesizer.go

```diff
@@ -229,71 +237,71 @@ func (t *targetGroupSynthesizer) SynthesizeSDKTargetGroups(ctx context.Context)
 			srvExport := &mcs_api.ServiceExport{}
 			if err := t.client.Get(ctx, srvExportName, srvExport); err == nil {
 
-				glog.V(6).Infof("Ignore TargetGroup(triggered by serviceexport) %v, %v since serviceexport object is found",
+				t.log.Infof("Ignore TargetGroup(triggered by serviceexport) %v, %v since serviceexport object is found",
 					*sdkTG.getTargetGroupOutput.Arn, *sdkTG.getTargetGroupOutput.Name)
 				continue
 			}
 		}
 
-		// if its parentref is HTTP/route, check the parent HTTPRoute exist
-		// Ignore if httpRoute does NOT exist
+		// if its parentRef is a route, check that the parent route exists
+		// Ignore if route does not exist
 		if *parentRef == latticemodel.K8SHTTPRouteType {
-			glog.V(6).Infof("TargetGroup %v, %v is referenced by HTTPRoute",
+			t.log.Infof("TargetGroup %v, %v is referenced by a route",
 				*sdkTG.getTargetGroupOutput.Arn, *sdkTG.getTargetGroupOutput.Name)
 
-			httpName, ok := tgTags.Tags[latticemodel.K8SHTTPRouteNameKey]
-			tgRouteName = *httpName
+			routeNameValue, ok := tgTags.Tags[latticemodel.K8SHTTPRouteNameKey]
+			tgRouteName = *routeNameValue
 
-			if !ok || httpName == nil {
-				glog.V(6).Infof("Ignore TargetGroup(triggered by httpRoute) %v, %v have no httproute name tag",
+			if !ok || routeNameValue == nil {
+				t.log.Infof("Ignore TargetGroup(triggered by route) %v, %v has no route name tag",
 					*sdkTG.getTargetGroupOutput.Arn, *sdkTG.getTargetGroupOutput.Name)
 				continue
 			}
 
-			httpNamespace, ok := tgTags.Tags[latticemodel.K8SHTTPRouteNamespaceKey]
+			routeNamespaceValue, ok := tgTags.Tags[latticemodel.K8SHTTPRouteNamespaceKey]
 
-			if !ok || httpNamespace == nil {
-				glog.V(6).Infof("Ignore TargetGroup(triggered by httpRoute) %v, %v have no httproute namespace tag",
+			if !ok || routeNamespaceValue == nil {
+				t.log.Infof("Ignore TargetGroup(triggered by route) %v, %v has no route namespace tag",
 					*sdkTG.getTargetGroupOutput.Arn, *sdkTG.getTargetGroupOutput.Name)
 				continue
 			}
 
-			httprouteName := types.NamespacedName{
-				Namespace: *httpNamespace,
-				Name:      *httpName,
+			routeName := types.NamespacedName{
+				Namespace: *routeNamespaceValue,
+				Name:      *routeNameValue,
 			}
 
-			httpRoute := &gateway_api.HTTPRoute{}
-
-			tgName := latticestore.TargetGroupName(*srvName, *srvNamespace)
-
-			if err := t.client.Get(ctx, httprouteName, httpRoute); err != nil {
-				glog.V(6).Infof("tgname %v is not used by httproute %v\n", tgName, httpRoute)
-
+			var route core.Route
+			if *sdkTG.getTargetGroupOutput.Config.ProtocolVersion == vpclattice.TargetGroupProtocolVersionGrpc {
```

Comment:
I would use switch in places where multiple versions supported

Reply:
This is on purpose, as we default to http with the sole exception of grpc